### PR TITLE
Feature/new test framework and added support for python3.10

### PR DESCRIPTION
**Pull request by VarunSharma15**
### Description
Added support for python 3.10.
Made changes to tests according to new test framework.

### Checklist
 - [ ] I have run this code in development and it appears to resolve the stated issue
 - [ ] This PR includes tests, or tests are not required/relevant for this PR
 - [ ] I have updated the `CHANGELOG.md` with information about my change

### CODE REVIEW

**Review by adamtwo — APPROVED** [review]

**Review by SatishChGit — APPROVED** [review]
Looks Good.

File: .github/workflows/ci-integration-tests.yml [hunk 1]
```diff
@@ -92,9 +92,9 @@ jobs:
           .logoff
           EOF
           # strip the random suffix from the pytest config
-          sed -i'.original' -e "s/_{{ var('_dbt_random_suffix') }}//" test/integration/*
+          sed -i'.original' -e "s/_{{ var('_dbt_random_suffix') }}//" tests/functional/*
```
[the remark this answers is not in the document]
Comment (VarunSharma15) [reply .github/workflows/ci-integration-tests.yml]: Made the changes.

File: tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+# The profile dictionary, used to write out profiles.yml
+# dbt will supply a unique schema per test, so we do not specify 'schema' here
+@pytest.fixture(scope="class")
+def dbt_profile_target():
+    return {
+        'type': 'teradata',
+        'threads': 1,
+        'host': 'localhost',
+        'user': 'dbc',
+        'password': 'dbc'
```
[the remark this answers is not in the document]
Comment (VarunSharma15) [reply tests/conftest.py]: Changes made to accommodate the env variables.

File: requirements_dev.txt [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
 wheel
-pytest~=6.0
+pytest~=7.0
 tox~=3.2
-pytest-dbt-adapter~=0.6.0
+dbt-tests-adapter~=1.1
 pylava~=0.3.0
-teradatasql>=16.20.0.0
-dbt-core==1.0.6
+teradatasql>=17.10.0.0
```
[the remark this answers is not in the document]
Comment (VarunSharma15) [reply requirements_dev.txt]: 16.20 is supported

File: tests/conftest.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (VarunSharma15) [reply tests/conftest.py]: Changes made to accommodate the env variables.

File: requirements_dev.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (VarunSharma15) [reply requirements_dev.txt]: 16.20 is supported

File: .bumpversion.cfg [hunk 4]
```diff
@@ -1,5 +1,5 @@
 [bumpversion]
-current_version = 1.0.0a
+current_version = 1.1.0a
```
[the remark this answers is not in the document]
Comment (VarunSharma15) [reply .bumpversion.cfg:2]: Replaced all of them.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,8 @@
-## dbt-teradata 1.0.0a
+## dbt-teradata 1.1.0a
 
 ### Features
+Conversion of tests aligned with new test framework
+Added support for python 3.10
```
[the remark this answers is not in the document]
Comment (VarunSharma15) [reply CHANGELOG.md:6]: Added.

**Pull request merged**